akkeeper/src/main/scala/akkeeper/config/RichConfig.scala

```diff
@@ -15,102 +15,133 @@
+  private def zookeeperConfig: Config = akkeeperConfig.getConfig("zookeeper")
+  private def restConfig: Config = akkeeperConfig.getConfig("api.rest")
+  private def kerberosConfig: Config = akkeeperConfig.getConfig("kerberos")
+  private def yarnConfig: Config = akkeeperConfig.getConfig("yarn")
+  private def yarnMaster: Config = yarnConfig.getConfig("master")
 
-  def getMapOfStrings(path: String): Map[String, String] = {
-    if (config.hasPath(path)) {
-      config.getConfig(path).entrySet().asScala.map(entry => {
-        entry.getKey -> entry.getValue.unwrapped().asInstanceOf[String]
-      }).toMap
+  // Akkeeper configs
+  def containers: Seq[ContainerDefinition] = {
+    if (akkeeperConfig.hasPath("containers")) {
+      val configContainers = akkeeperConfig.getConfigList("containers").asScala
+      configContainers.map(ContainerDefinition.fromConfig)
     } else {
-      Map.empty
+      Seq.empty
     }
   }
 
-  def getListOfStrings(path: String): Seq[String] = {
-    if (config.hasPath(path)) {
-      config.getStringList(path).asScala
+  // Akka configs
+  def akkaActorSystemName: String = akkaConfig.getString("system-name")
+  def akkaPort: Int = akkaConfig.getInt("port")
+  def akkaSeedNodesNum: Int = akkaConfig.getInt("seed-nodes-num")
+  def akkaJoinClusterTimeout: FiniteDuration = akkaConfig.getDuration("join-cluster-timeout")
+  def akkaLeaveClusterTimeout: FiniteDuration = akkaConfig.getDuration("leave-cluster-timeout")
+
+  // Monitoring configs
+  def monitoringLaunchTimeout: FiniteDuration = monitoringConfig.getDuration("launch-timeout")
+
+  // Launcher configs
+  def launcherTimeout: Option[Duration] = {
+    if (launcherConfig.hasPath("timeout")) {
+      Some(launcherConfig.getDuration("timeout", TimeUnit.SECONDS).seconds)
     } else {
-      Seq.empty
+      None
     }
   }
 
-  def getActorSystemName: String = {
-    config.getString("akkeeper.akka.system-name")
-  }
+  // Zookeeper configs
+  def zookeeperClientConfig: ZookeeperClientConfig = ZookeeperClientConfig.fromConfig(zookeeperConfig)
 
-  def getYarnConfig: Config = {
-    config.getConfig("akkeeper.yarn")
-  }
+  // Rest API configs
+  def restPort: Int = restConfig.getInt("port")
+  def restPortMaxAttempts: Int = restConfig.getInt("port-max-attempts")
+  def restRequestTimeout: Timeout = Timeout(
+    restConfig.getDuration("request-timeout", TimeUnit.MILLISECONDS),
+    TimeUnit.MILLISECONDS)
 
-  def getYarnApplicationName: String = {
-    getYarnConfig.getString("application-name")
-  }
+  // Kerberos configs
+  def kerberosTicketCheckInterval: Long =
+    kerberosConfig.getDuration("ticket-check-interval", TimeUnit.MILLISECONDS)
 
-  def getYarnStagingDirectory(conf: Configuration, appId: String): String ={
+  // Yarn configs
+  def yarnApplicationName: String = yarnConfig.getString("application-name")
+  def yarnJvmArgs: mutable.Seq[String] = yarnConfig.getStringList("master.jvm.args").asScala
```

Comment:
This doesn't belong to `yarn` settings and should be `masterJvmArgs` instead.

Reply:
Why is it mutable Seq? This approach would expose this configuration property to updates from outside.

Reply:
Done